solvers/chuffed/build.rs

```diff
@@ -0,0 +1,80 @@
+// adapted from
+// https://github.com/gokberkkocak/rust_glucose/blob/master/build.rs
+// and
+// 1https://rust-lang.github.io/rust-bindgen/non-system-libraries.html
+
+use std::process::Command;
+use std::env;
+use std::path::PathBuf;
+
+fn main() {
+    println!("cargo:rustc-rerun-if-changed=vendor");
+    println!("cargo:rerun-if-changed=build.rs");
+
+    // must be ./ to be recognised as relative path
+    // from project root
+    println!("cargo:rustc-link-search=all=./solvers/chuffed/vendor/build/");
+    println!("cargo:rustc-link-lib=static=chuffed");
+
```

Comment:
I've added this along with your suggested changes to lib.rs. I've also tried to run your example main function after putting createVar into the allow list in build.rs. I get a different error from you when trying to compile now even after adding the link to `libchuffed_fzn.a`. Were you able to get your example main running after you made those changes?

Reply:
Could you send the error that you are getting so that I can look into it? I was able to get it to run on my end.

Reply:
> Could you send the error that you are getting so that I can look into it? I was able to get it to run on my end.

Yeah sure thing, it's quite a lot so just gonna show a snippet

```
error: linking with `cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:/home/kf77/.opam/default/bin:/home/kf77/local/bin:/home/kf77/local/bin:/home/kf77/local/bin:/home/kf77/local/bin:/home/kf77/.cargo/bin:/home/kf77/.cabal/bin:/home/kf77/.ghcup/bin:/home/kf77/.local/bin:/home/kf77/bin:/usr/share/Modules/bin:/usr/local/julia/bin:/home/kf77/usr/bin:/cs/home/kf77/usr/bin:/usr/local/maven/bin:/usr/local/amazon-corretto-17/bin:/usr/lib64/ccache:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/home/kf77/.dotnet/tools" VSLANG="1033" "cc" "-m64" "/tmp/rustcyP3VVJ/symbols.o" "/home/kf77/conjure-oxide/target/debug/deps/chuffed_rs-3916e90f286e3739.38o4ig1bt7kkyjdv.rcgu.o" "/home/kf77/conjure-oxide/target/debug/deps/chuffed_rs-3916e90f286e3739.3po8wt4hdebzumrt.rcgu.o" "/home/kf77/conjure-oxide/target/debug/deps/chuffed_rs-3916e90f286e3739.4jresi8mttp46v6o.rcgu.o" "/home/kf77/conjure-oxide/target/debug/deps/chuffed_rs-3916e90f286e3739.4jyq2iwe1v1mymye.rcgu.o" "/home/kf77/conjure-oxide/target/debug/deps/chuffed_rs-3916e90f286e3739.520za8226h1pu3p6.rcgu.o" "/home/kf77/conjure-oxide/target/debug/deps/chuffed_rs-3916e90f286e3739.5brh85kw9qop90tt.rcgu.o" "-Wl,--as-needed" "-L" "/home/kf77/conjure-oxide/target/debug/deps" "-L" "./solvers/chuffed/vendor/build/" "-L" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/home/kf77/conjure-oxide/target/debug/deps/libchuffed_rs-d37182f69c381980.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-b850d2b001350814.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-a3d926e3ce38f6cc.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libobject-0a660adf51e0d01c.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libmemchr-49252de8bfa771e2.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libaddr2line-6f52a21444f37c70.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgimli-57e8575f0ca731a2.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-da4065b9a8ae7be4.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd_detect-1ad47776352198ff.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libhashbrown-7075f8a8dc932a7c.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-a494eb4ddbb000f0.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libminiz_oxide-672778ed8628861b.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libadler-9bba1c59dcc21440.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-b9849fb628ce12eb.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-0ea4fcc08eb96f77.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-a8ba743c059cc198.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-e4195ee2443e1656.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-04991604e2730fd6.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-f02db372677d4667.rlib" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-448eba810c83029d.rlib" "-Wl,-Bdynamic" "-lstdc++" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/home/kf77/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/home/kf77/conjure-oxide/target/debug/deps/chuffed_rs-3916e90f286e3739" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs"
  = note: /usr/bin/ld: /home/kf77/conjure-oxide/target/debug/deps/libchuffed_rs-d37182f69c381980.rlib(modelling.cpp.o): relocation R_X86_64_32 against symbol `sat' can not be used when making a PIE object; recompile with -fPIE
          /usr/bin/ld: /home/kf77/conjure-oxide/target/debug/deps/libchuffed_rs-d37182f69c381980.rlib(bool-view.cpp.o): relocation R_X86_64_32 against `.LC4' can not be used when making a PIE object; recompile with -fPIE
          /usr/bin/ld: /home/kf77/conjure-oxide/target/debug/deps/libchuffed_rs-d37182f69c381980.rlib(sat.cpp.o): relocation R_X86_64_32 against `.LC3' can not be used when making a PIE object; recompile with -fPIE
          /usr/bin/ld: /home/kf77/conjure-oxide/target/debug/deps/libchuffed_rs-d37182f69c381980.rlib(conflict.cpp.o): relocation R_X86_64_32 against `.LC1' can not be used when making a PIE object; recompile with -fPIE
          /usr/bin/ld: /home/kf77/conjure-oxide/target/debug/deps/libchuffed_rs-d37182f69c381980.rlib(int-var.cpp.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIE
          /usr/bin/ld: /home/kf77/conjure-oxide/target/debug/deps/libchuffed_rs-d37182f69c381980.rlib(int-var-el.cpp.o): relocation R_X86_64_32 against `.LC3' can not be used when making a PIE object; recompile with -fPIE
          /usr/bin/ld: /home/kf77/conjure-oxide/target/debug/deps/libchuffed_rs-d37182f69c381980.rlib(int-var-sl.cpp.o): relocation R_X86_64_32S against symbol `_ZN10Propagator6wakeupEii' can not be used when making a PIE object; recompile with -fPIE
          /usr/bin/ld: /home/kf77/conjure-oxide/target/debug/deps/libchuffed_rs-d37182f69c381980.rlib(int-var-ll.cpp.o): relocation R_X86_64_32 against `.LC0' can not be used when making a PIE object; recompile with -fPIE
          /usr/bin/ld: /home/kf77/conjure-oxide/target/debug/deps/libchuffed_rs-d37182f69c381980.rlib(ldsb.cpp.o): relocation R_X86_64_32 against symbol `sat' can not be used when making a PIE object; recompile with -fPIE
```

Reply:
For history's sake, this is due to the issues with AlmaLinux discussed here: https://github.com/conjure-cp/conjure-oxide/pull/6#issuecomment-1745675358